jwst/cube_build/blot_cube_build.py

```diff
@@ -208,24 +194,34 @@ def blot_images(self):
 
                 ycenter = np.arange(ysize)
                 xcenter = np.arange(xsize)
+                self.ycenter_grid, self.xcenter_grid = np.mgrid[0:ysize,
+                                                               0:xsize]
+
                 # for NIRSPEC each file has 30 slices
                 # wcs information accessed seperately for each slice
 
                 nslices = 30
                 log.info('Looping over 30 slices on NIRSPEC detector, this takes a little while')
                 for ii in range(nslices):
+                    # for each slice pull out the blotted values that actually fall on the slice region
+                    # used the bounding box of each slice to determine the slice limits
                     slice_wcs = nirspec.nrs_wcs_set_input(model, ii)
                     x_slice, y_slice = slice_wcs.invert(self.cube_ra, self.cube_dec, self.cube_wave)
-                    # there could be values that are nan because they do not fall on this slice.
                     x_slice = np.ndarray.flatten(x_slice)
                     y_slice = np.ndarray.flatten(y_slice)
                     flux_slice = np.ndarray.flatten(self.cube_flux)
-                    # print('number of original cube values mapped',flux_slice.size)
-                    valid = ~np.isnan(y_slice)
-                    x_slice = x_slice[valid]
-                    y_slice = y_slice[valid]
-                    flux_slice = flux_slice[valid]
-                    # print('number of valid cube values mapped',flux_slice.size)
+                    # print('number of original values', flux_slice.size)
```

Comment:
Regarding the commented `print` statements: Much as the previous comment, these should be either removed or clarified. Helps with readability.